### ISSUE TITLE
Automatic timezone update

### ISSUE THREAD

**Issue by mardy** [ubports/ubuntu-touch#379]
### Description of the feature
When travelling to another country, it would be nice if the timezone were automatically updated. In that way, the time would be correct.

### Implementation brainstorming
If one or two sims are in the device, the timezone information could be get from the SIM, by using this D-Bus call:
```
dbus-send --system --print-reply --type=method_call --dest=org.ofono /ril_0 org.ofono.NetworkTime.GetNetworkTime
```
This returns a `Timezone` field with the number of seconds of offset from UTC. However, not all mobile operators give out this information, so the implementation must be able to handle the case where no information is provided.
Other ways to get this information might be via the GPS, or via geolocating the IP address of the current connection.

In all cases, this is fragile and should be optional: maybe the system settings could have an "automatic time zone" option, which can be toggled on and off like the one about network time update.

**Comment by Flohack74** [comment]
Or, as some mobile phones do, present a popup "Timezone change detected, please confirm its xx:yy" ?

**Comment by mihaelmilea** [comment]
I subscribe.

**Comment by mateosalta** [comment]
I  always wondered if there could be city specific times shown in the time/date pull down - similar to the clock app. If traveling between two time zones often, this could be useful. but would like it to display mainly the timezone I'm in, or quickly see the time it will be, to judge how much time needed to get there.

**Comment by doniks** [comment]
@mateosalta I agree that seeing multiple time zones at once would be a nice feature. Maybe it could be an app or a scope or smt else. Either way, it is definitely a separate problem and should be raised as a separate issue.

**Comment by JassMan23** [comment]
@doniks I also agree with seeing multiple time zones. I had a Tizen phone before an unfortunate back-pocket incident, and one of the things that impressed me was when roaming it showed "Home" and "Local" times on the welcome screen. The clock always showed "local" time since when you are at home it is local.

I believe that even if networks don't show a time offset, the country can still be extracted from the current network ID.
From Wikipedia:
A mobile country code (MCC) is used in combination with a mobile network code (MNC) (a combination known as an "MCC/MNC tuple") to uniquely identify a mobile network operator (carrier) using the GSM (including GSM-R), UMTS, and LTE public land mobile networks. Some but not all CDMA, iDEN, and satellite mobile networks are identified with an MCC/MNC tuple as well.
